File: python/7 kyu/214.py
# https://www.codewars.com/kata/5b72970db3d977e0f9000030/train/python

# Passed

def crossedwords(str1, str2):
    for letter in str2:
        if letter in str1:
            col_idx = str1.index(letter)
            row_idx = str2.index(letter)
            break
            
    crossword_rows = []
    for idx, char in enumerate(str2):
        if idx != row_idx:
            crossword_row = [" "] * len(str1)
            crossword_row[col_idx] = str2[idx]
        else:
            crossword_row = list(str1)
            
        crossword_rows.append(crossword_row)
    
    crossword_representation = "\n".join([''.join(row) for row in crossword_rows]) + "\n"
    return crossword_representation

output = crossedwords('GRAPHICAL', 'SYNTHESIS')
print(output)

File: nodejs/7 kyu/214.js
// https://www.codewars.com/kata/5806445c3f1f9c2f72000031/train/javascript

// Passed

function meanVsMedian(numbers) {
  numbers = numbers.sort((a, b) => a - b);
  let mean = numbers.reduce((prev, curr) => prev + curr, 0) / numbers.length;
  let idx, median;
  if (numbers.length % 2 === 0) {
    idx = numbers.length / 2;
    median = (numbers[idx] + numbers[idx + 1]) / 2;
  } else {
    idx = Math.floor(numbers.length / 2);
    median = numbers[idx];
  }

  if (mean > median) {
    return "mean";
  } else if (median > mean) {
    return "median";
  } else {
    return "same";
  }
}

const output = meanVsMedian([1, 2, 37]);
console.log(output);